easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -553,21 +553,29 @@ def update(self, key, value, allow_duplicate=True):
         """
         Update a string configuration value with a value (i.e. append to it).
         """
+        if isinstance(value, string_type):
+            lval = [value]
+        elif isinstance(value, list):
+            lval = value
+        else:
+            msg = "Can't update configuration value for %s, because the "
+            msg += "attempted update value, '%s', is not a string or list."
+            raise EasyBuildError(msg, key, value)
+
         prev_value = self[key]
         if isinstance(prev_value, string_type):
-            if allow_duplicate or value not in prev_value:
-                self[key] = '%s %s ' % (prev_value, value)
+            for item in lval:
+                if allow_duplicate or (not re.search(r'(^|\s+)%s(\s+|$)' % re.escape(item), prev_value)):
```

Comment:
I think the `re.search` deserves a clarifying comment above? Something like:

"# only add value to string if it's not there yet (surrounded by whitespace)